gpt4all-chat/chatllm.cpp

```diff
@@ -280,19 +280,21 @@ bool ChatLLM::loadModel(const ModelInfo &modelInfo)
     m_llModelInfo.fileInfo = fileInfo;
 
     if (fileInfo.exists()) {
-        if (isChatGPT) {
+        if (isOnline) {
             QString apiKey;
-            QString chatGPTModel = fileInfo.completeBaseName().remove(0, 8); // remove the chatgpt- prefix
             {
                 QFile file(filePath);
                 file.open(QIODeviceBase::ReadOnly | QIODeviceBase::Text);
                 QTextStream stream(&file);
                 apiKey = stream.readAll();
                 file.close();
             }
-            m_llModelType = LLModelType::CHATGPT_;
-            ChatGPT *model = new ChatGPT();
-            model->setModelName(chatGPTModel);
+            m_llModelType = LLModelType::API_;
+            ChatAPI *model = new ChatAPI();
+            QString path = modelInfo.filename();
+            path.chop(4);
+            model->setModelName(path);
```

Comment:
Guessing this is what you mean about model name? Ok, yeah, I don't like this. I think what we likely want to be doing here is getting the key model name from the file itself. We should probably come up with a new format for the file and/or version it. My preference would be a list of key value pairs as I doubt we'll need more structure than that?

Reply:
If we want to support a fixed set of model types, then I think we should have a hardcoded, strict bidirectional mapping between model names and files that represent them. If we want to support *arbitrary* OpenAI-compatible models, we *could* normalize and lowercase the name of the model to use as part of the filename (I think it should still have a gpt4all prefix), but I agree that the name should be stored inside of the file.